test/e2e/util.go

```diff
@@ -50,29 +56,25 @@ func Failf(format string, a ...interface{}) {
 	Fail(fmt.Sprintf(format, a...), 1)
 }
 
-func waitForPodRunning(c *client.Client, id string, tryFor time.Duration) error {
-	trySecs := int(tryFor.Seconds())
-	for i := 0; i <= trySecs; i += 5 {
-		time.Sleep(5 * time.Second)
-		pod, err := c.Pods(api.NamespaceDefault).Get(id)
+func waitForPodRunning(c *client.Client, podName string) error {
+	By(fmt.Sprintf("waiting up to %v for pod %s status to be running", podStartTimeout, podName))
+	for start := time.Now(); time.Since(start) < podStartTimeout; time.Sleep(5 * time.Second) {
+		pod, err := c.Pods(api.NamespaceDefault).Get(podName)
 		if err != nil {
-			return fmt.Errorf("Get pod %s failed: %v", id, err.Error())
+			return fmt.Errorf("Get pod %s failed: %v", podName, err.Error())
 		}
 		if pod.Status.Phase == api.PodRunning {
 			return nil
 		}
-		Logf("Waiting for pod %s status to be %q (found %q) (%d secs)", id, api.PodRunning, pod.Status.Phase, i)
+		Logf("Waiting for pod %s status to be %q (found %q) (%d secs)", podName, api.PodRunning, pod.Status.Phase, time.Since(start).Seconds())
 	}
-	return fmt.Errorf("Gave up waiting for pod %s to be running after %d seconds", id, trySecs)
+	return fmt.Errorf("Gave up waiting for pod %s to be running after %v", podName, podStartTimeout)
 }
 
-// waitForPodNotPending returns false if it took too long for the pod to go out of pending state.
-func waitForPodNotPending(c *client.Client, ns, podName string, tryFor time.Duration) error {
-	trySecs := int(tryFor.Seconds())
-	for i := 0; i <= trySecs; i += 5 {
-		if i > 0 {
-			time.Sleep(5 * time.Second)
-		}
+// waitForPodNotPending returns an error if it took too long for the pod to go out of pending state.
+func waitForPodNotPending(c *client.Client, ns, podName string) error {
```

Comment:
This func looks like an almost exact dupe of the previous one with a different success condition. Can you consolidate the shared logic into a helper function?